framework/aster-frame/src/arch/x86/trap.rs

```diff
@@ -47,6 +48,40 @@ extern "sysv64" fn trap_handler(f: &mut TrapFrame) {
                 handle_virtual_exception(f, &ve_info);
             }
             &PAGE_FAULT => {
+                let page_fault_addr = unsafe { x86::controlregs::cr2() };
+
+                // Handle page fault from user space.
+                if (0..MAX_USERSPACE_VADDR).contains(&page_fault_addr) {
+                    let current_task = current_task().unwrap();
```

Comment:
In the conditional block there is a long process till the `return` in line 82 that could be moved into a function instead.